### ISSUE TITLE
rhg_compute_tools.gcs.cp might not always do what we expect

### ISSUE THREAD

**Issue by bolliger32** [RhodiumGroup/rhg_compute_tools#86]
I'm not 100% positive I wrote the `cp` command correctly to handle cases where you are copying directories to other directories. [The rules](https://cloud.google.com/storage/docs/gsutil/commands/cp) for exactly how the copy is conducted, depending on whether or not you specify `-r`, whether you are using single or double wildcard characters, and whether you are copying files or directories, are quite complicated. It's making my head spin and so I gave up trying to fix things for now. `sync` is often a simpler and easier option b/c you know you are syncing a directory to a directory. But at some point, we should take a look at `cp` and make sure it's working appropriately.

**Comment by brews** [comment]
Can you outline the behavior we want with some concrete examples here, @bolliger32?

**Comment by bolliger32** [comment]
Yeah, will do when I get some time. In general, the way the others work is to first call the appropriate `gsutil` command and then create all the necessary directories in gcsfuse so that you are actually able to see the files (b/c gsutil won't do that). It's just a little more complicated with cp than with sync b/c you could be copying either folders or files and you can use wildcards, etc. I'll put some examples up when i get a chance
